lib/optimizer/src/Transformer/ServerSideRendering.php

```diff
@@ -978,4 +982,26 @@ private function isExcludedElement(DOMElement $element)
     {
         return in_array($element->tagName, self::EXCLUDED_ELEMENTS, true);
     }
+
+    /**
+     * Check if a number is zero.
+     *
+     * This works correctly with both integer and float values.
+     *
+     * @param int|float $number Number to check for zero.
+     * @return bool Whether the provided number is zero.
+     * @throws InvalidArgument When an unsupported number type is provided.
+     */
+    private function isZero($number)
+    {
+        if (is_int($number)) {
+            return $number === 0;
+        }
+
+        if (!is_float($number)) {
+            throw InvalidArgument::forNumericComparison($number);
+        }
+
+        return abs($number - 0.0) < self::FLOATING_POINT_EPSILON;
```

Comment:
Why subtract `0.0`? Is this for PhpStan?

Reply:
It is the blind application of an algorithm for comparing floating point numbers: `abs( $a - $b ) < $epsilon`. I was not 100% sure whether subtracting a float `0.0` would have an actual effect on the result or not.

However, we're probably safe removing the subtraction, I can't think of a way it would break without it, as we're already checking whether `$number` is a `float` to begin with.

Reply:
Right, subtracting `0.0` would convert it into a float, which it already is.

Reply:
OK, then to prevent confusion later:

```suggestion
        return abs($number) < self::FLOATING_POINT_EPSILON;
```